src/de/schildbach/pte/NetworkId.java

```
@@ -25,7 +25,7 @@ public enum NetworkId {
     RT,
 
     // Germany
-    DB, BVG, VBB, NVV, BAYERN, MVV, INVG, AVV_AUGSBURG, VGN, VVM, VMV, SH, GVH, BSVAG, VBN, NASA, VMT, VVO, VGS, VRR, VRS, AVV_AACHEN, MVG, VRN, VVS, DING, KVV, NVBW, VVV,
+    DB, DBMOVAS, BVG, VBB, NVV, BAYERN, MVV, INVG, AVV_AUGSBURG, VGN, VVM, VMV, SH, GVH, BSVAG, VBN, NASA, VMT, VVO, VGS, VRR, VRS, AVV_AACHEN, MVG, VRN, VVS, DING, KVV, NVBW, VVV,
```

Given the fact that the HCI API now seems to be shut off, I'm now leaning towards replacing the old ID. If on the other hand there is a chance the old API will be ressurected, I'd keep it separate. What do you think?

Sorry for the back and forth about this one.

---

I don't think that the API will return because 1) many things have hinted at it being turned off 2) people from within DB have said that this would probably happen 3) it just makes sense for them to turn it off, and at this point in time after coming back from the christmas holidays. But ofc I might be wrong.
Let me know or just change it yourself. Anyways it's probably a good idea if you quickly test the forward compatibility of the serialized locations and journeys etc from Db to DbMovas. I can't do that anymore, since I had to uninstall due to signing and have barely anything in my Oeffi search history for the old DB Provider and for obvious reasons can't create any new entries now...

---

You could also keep the old one around as `DB_HAFAS` or sth and mark it as deprecated in the Oeffi UI, so that in the worst case one can still select it, albeit without saved journeys.

---

Newbie question: Does labelling the new client as `DB` have any benefits?

---

Would reusing the old ID avoid the need for users to switch their network in the settings?

In this case this would be, in my opinion, a strong reason to do it, so that for users after an update everything works again as they are used to without the need for an manual procedure.

---

>> Would reusing the old ID avoid the need for users to switch their network in the settings?
> 
> In this case this would be, in my opinion, a strong reason to do it, so that for users after an update everything works again as they are used to without the need for an manual procedure.

I'd consider that a responsibility of the Öffi code, not `public-transport-enabler`'s, but maybe building that into Öffi is not worth the complexity.

---

Both reasonings are valid I'd say. In the past, if an API has a designated successor *and* IDs are compatible I (more or less transparently) replaced it at PTE level.

---

> You could also keep the old one around as `DB_HAFAS` or sth and mark it as deprecated in the Oeffi UI, so that in the worst case one can still select it, albeit without saved journeys.

I'd rather not do that, as it seems to accumulate all the downsides: needs migration code in Öffi, adds another (sooner or later redundant) ID. To ressurect the old API if needed, git would be our friend I guess.

---

The diff between the two tests shows that some changes are necessary,
1) different constructor
2) unsupported POI search
3) station `692990` that only very rarely has departures and thus fails frequently due to the 1 hour limit (on that note, don't forget the `selectedDepartures.size()+1` fix in Oeffi, it's important for stops where there is one departure per hour)
4) changed id of München - Berg am Laim (don't recall exactly)
5) fix some evergreens/fail early with additional asserts
6) an additonal test checking querying for stations only by name and with backwards compatible location address/poi ids
6) some changed errors, because for too close stations the new API just returns an empty result
So rather keep the new Test or feel free to check everything again if you want :)